packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/evaluate-expression.ts

```diff
@@ -111,13 +118,18 @@ function createEvalFn(runtime: lambda.Runtime | undefined, scope: Construct) {
     [NO_RUNTIME]: '41256dc5-4457-4273-8ed9-17bc818694e5',
   };
 
-  const uuid = nodeJsGuids[runtime?.name ?? NO_RUNTIME];
+  let uuid = nodeJsGuids[props.runtime?.name ?? NO_RUNTIME];
   if (!uuid) {
-    throw new Error(`The runtime ${runtime?.name} is currently not supported.`);
+    throw new Error(`The runtime ${props.runtime?.name} is currently not supported.`);
+  }
+
+  if (props.role) {
+    uuid += props.role.node.addr;
```

Comment:
I am not totally sure if this is the right way to go about this. But the point is, I want the role to be part of the signature of the function created, so if I want to use different roles in one part of the stack or don't set a role at all, it will create different functions, instead of the first occurrence blocking all of the others.

Reply:
Is there a risk of this uuid being too long?

Reply:
coI could not find one documented. Also, addr should have a fixed length of 42. SingletonFunction eventually passes it to `tryFindChild`. So that is where a length constraint could potentially come up, but otherwise, I think the risk is very low/none.

Reply:
More likely, what I'm worried about is this addr value changing between deployments and deleting/recreating the function. I'm not sure I understand how the value of addr is calculated well enough to guarantee that won't happen

Reply:
What even is the value of `props.role.node.addr`? Could we use the `props.role.roleName/roleArn` instead?

Reply:
> What is the value of props.role.node.addr?

https://docs.aws.amazon.com/cdk/api/v2/docs/constructs.Node.html#addr

tl:dr; SHA-1 of the components of the construct path.

> Could we use the props.role.roleName/roleArn instead?

No. this could be a token and change between deployments.

Reply:
Were you able to test if the value of `props.role.node.addr` changes between deployments? From the description, `Addresses are calculated using a SHA-1 of the components of the construct path.`,  I'm not confident that it would be the same between deployments.